elastic/kibana
# [Dashboard] [Controls] Remove options list `"Allow <x>"` toggles

Closes https://github.com/elastic/kibana/issues/147140

## Summary

After discussion surrounding the "author" versus "analyst" experience for the controls, we have come to the conclusion that a simpler create experience is necessary to streamline our current process. This means removing **all** options that are not absolutely necessary - specifically, this PR removes the UX for the following "Additional settings" toggles from the creating/editing flyout:

- Allow selections to be excluded (`hideExclude`)
- Allow exists query (`hideExists`)
- Allow dynamic sorting of suggestions (`hideSort`)

This will leave only two additional settings: 
- Allow multiple selections in dropdown
- Ignore timeout for results

While the UX for the removed toggles will be removed, the logic is kept - this means that, for the Control Group API that solutions will be using, this functionality could still be exposed to the consumers.


### Migration 
Consider a scenario where, in 8.6, a user creates an options list control with `hideExists` set to `true`- that is, the `Exists`  option is not displayed in the options list suggestions. After upgrading to 8.7, they realize that they want to turn the `Exists` query back on - but they can't, because the toggle has been removed so `hideExists` is stuck as `true`! Their only choice is to remove the control and start over. 

In order to avoid this scenario, I added an 8.7 migration that goes through all existing control group panels and, if a control is an options list, it removes the `hideExclude` and `hideExists` keys from the explicit input. This is effectively the same was setting these to `false` since they are optional.

> **Note**
> Because the `hideExclude` and `hideExists` toggles were added back in 8.6.0 but `hideSort` was only added in 8.7.0 (which has not yet reached feature freeze), that is why only the `hideExclude` and `hideExists` keys are removed. There is no need to add a migration for `hideSort` because there will be no customer impact from removing the "Allow dynamic sorting of suggestions" toggle.


## How to test

For the sake of convenience, I have exported various dashboard saved objects from 8.6 (all of which used the demo "Kibana Sample Data eCommerce" data view) in order to test the migration to 8.7:
- `8.6_OneControlNoMigrations.ndjson`
    This dashboard had a single options list control that should require **no** migration because it was created with the default settings:

    ![image](https://user-images.githubusercontent.com/8698078/206319266-84377dd8-8e53-4680-81f3-60e7b33ba15e.png)


- `8.6_OneControlTogglesOff.ndjson`
    This dashboard had a single options list control with **every single** toggle turned off:

    ![image](https://user-images.githubusercontent.com/8698078/206319041-bdb8d28f-4f70-4ac9-a2d2-87600b0db66b.png)


- `8.6_TwoOptionsListControls.ndjson`
    This dashboard had two options list controls with the following settings:
    
    ![TwoOptionsListControls](https://user-images.githubusercontent.com/8698078/206320381-4aad9c78-0bb0-4d30-b7f7-3dc3e6b7b2a2.png)
 
    The first control also had the `exists` query selected with `exclude` set to `true`, while the second control had non-default size settings, and I made some selections and saved them as part of the dashboard.

- `8.6_MultipleControlTypes.ndjson`
    This dashboard had four controls: two options list controls with the following settings, a range slider control, and a time slider control:

     ![MultipleControlTypes](https://user-images.githubusercontent.com/8698078/206321542-8a9943ce-bf87-4de7-94fe-7ee370b92a23.png)


These can each be found in the following ZIP file:
> **Warning**
> [DashboardSavedObjectsForTestingMigration.zip](https://github.com/elastic/kibana/files/10180945/DashboardSavedObjectsForTestingMigration.zip)


In order to test this PR, you should (at minimum) download the previous ZIP, individually import each dashboard into Kibana 8.7, and ensure that:
1. the resulting JSON looks as expected, with no `hideExists` or `hideExclude` keys present, and
2. the dashboard loads correctly and each options list control has the ability to include/exclude, sort, and create an exists query regardless of the previous state of `hideExists` or `hideExclude`


### Checklist



### For maintainers
